src/dbnode/encoding/m3tsz/iterator.go

```diff
@@ -173,40 +180,37 @@ func (it *readerIterator) readIntSigMult() {
 }
 
 func (it *readerIterator) readIntValDiff() {
+	// read both sign bit and digits in one read
+	bits := it.readBits(it.sig + 1)
 	sign := -1.0
-	if it.readBits(1) == opcodeNegative {
+	if (bits >> it.sig) == opcodeNegative {
 		sign = 1.0
 	}
-
-	it.intVal += sign * float64(it.readBits(it.sig))
+	bits &= ^uint64(1 << it.sig)
```

Comment:
the goal was to clear the opcode bit, but it only needs to be cleared if it was set in the first place, but since we already have a conditional, can just do a XOR there. thanks!